lib/view_models/stream_view_model.dart

```diff
@@ -25,52 +25,82 @@ class StreamViewModel extends StateNotifier<StreamState> {
     }
   }
 
-  Future<void> fetchStream(Int64 streamId) async {
-    _logger.i('Fetching stream');
-    state = state.copyWith(isLoading: true);
+  /// handles fetching of all thumbnails live or not
+  Future<void> fetchThumbnails() async {
+    List<String> fetchedThumbnails = [];
+    if (state.streams == null) {
+      return;
+    } else {
+      for (var stream in state.streams!) {
+        // Add null assertion here
+        try {
+          String thumbnail;
+          if (stream.liveNow) {
+            thumbnail = await fetchStreamThumbnail(stream.id);
+          } else {
+            thumbnail = await fetchVODThumbnail(stream.id);
+          }
+          fetchedThumbnails.add(thumbnail);
+        } catch (e) {
+          _logger.e('Error fetching thumbnail for stream ID ${stream.id}: $e');
+          fetchedThumbnails.add(
+            '/thumb-fallback.png',
+          ); // Add a default thumbnail path in case of error
+        }
+      }
+
+      state = state.copyWith(thumbnails: fetchedThumbnails, isLoading: false);
+    }
+  }
+
+  /// Handles Fetching of a thumbnail for a live stream
+  Future<String> fetchStreamThumbnail(Int64 streamId) async {
     try {
-      final stream = await StreamHandler(_grpcHandler).fetchStream(streamId);
-      state = state.copyWith(streams: [stream], isLoading: false);
+      _logger.i('Fetching thumbnail VOD for stream ID: $streamId');
+      String thumbnailPath =
+          await StreamHandler(_grpcHandler).fetchThumbnailVOD(streamId);
```

Comment:
should be like this?

```suggestion
          await StreamHandler(_grpcHandler).fetchThumbnailStreams(streamId);
```